backend/pkg/runtime_scanner/scanner.go

```diff
@@ -138,58 +143,57 @@ func (s *RuntimeScanner) setScanConfig(scanConfig *ScanConfig) {
 func (s *RuntimeScanner) startScan(scanConfig *ScanConfig) error {
 	startTime := time.Now().UTC()
 	s.lock.Lock()
+	if s.stopCurrentScanChan != nil {
+		return fmt.Errorf("found existing cancel channel which may mean a scan is already running, call StopCurrentScan() before calling startScan again to ensure any existing scan is aborted and the cancel channel is cleaned up")
+	}
 	s.lastScanStartTime = startTime
-	stop := s.stopCurrentScanChan
 	s.scannedNamespaces = scanConfig.Namespaces
+	s.stopCurrentScanChan = make(chan struct{})
```

Comment:
I think that it is better to save it locally on
```
stop := s.stopCurrentScanChan
```
since it is protected with a lock

Reply:
I actually choose not to save it locally because we only ever want to have one scan running, if stopCurrentScanChan is already set and the RuntimeScanner tries to set it (its an internal variable so should only be changed by the RuntimeScanner) then it means there is already a scan running and the StopCurrentScan function should be called to clear it.

I was trying to decide whether the startScan function should hold a read lock the entire time its waiting on s.stopCurrentScanChan to prevent a writer from replacing it behind its back

Reply:
I know that we only have one scan running, but since we closing and replacing the channel under a lock I think it will be safer to save it locally here to avoid holding a read lock the entire time it's waiting on it.

Reply:
updated it so that we've got a nice function for allocating a new stop channel that should always be used to write stopCurrentScanChan and it return the allocated channel, then I can do a defer on the StopCurrentChan for cleanup when the scan is completed.